4_processing_features/scripts/plate1_extract_sc_cp.py

```diff
@@ -0,0 +1,226 @@
+
+# set paths for pybasic cellpose method
+method3 = "pybasic_cellpose"
+sql_file3 = f"{method3}.sqlite"
+single_cell_file3 = f"sqlite:///{cp_dir}/Analysis_Output/Plate1_Output/{sql_file3}"
+
+# set paths for cellprofiler cellpose method
+method4 = "cellprofiler_cellpose"
+sql_file4 = f"{method4}.sqlite"
+single_cell_file4 = f"sqlite:///{cp_dir}/Analysis_Output/Plate1_Output/{sql_file4}"
+
+# set path to the platemap for plate 1
+platemap_file = f"{cp_dir}/Metadata/platemap_NF1_CP.csv"
+
+
+# ## Set up names for linking columns between tables in the database file
+
+# In[4]:
+
+
+# Define custom linking columns between compartments
+linking_cols = {
+    "Per_Cytoplasm": {
+        "Per_Cells": "Cytoplasm_Parent_Cells",
+        "Per_Nuclei": "Cytoplasm_Parent_Nuclei",
+    },
+    "Per_Cells": {"Per_Cytoplasm": "Cells_Number_Object_Number"},
+    "Per_Nuclei": {"Per_Cytoplasm": "Nuclei_Number_Object_Number"},
+}
+
+
+# ## Load in platemap
+
+# In[5]:
+
+
+# Load platemap file
+platemap_df = pd.read_csv(platemap_file)
+platemap_df
+
+
+# ## Perform extraction with All CellProfiler method
+# 
+# - Merge single cells
+# - Normalize
+# - Feature selection on normalized features
+
+# In[6]:
+
+
+sc_util.extract_single_cells(
+    single_cell_file=single_cell_file1,
+    linking_cols=linking_cols,
+    platemap_df=platemap_df,
+    output_folder=output_dir,
+    method_name=method1,
+    norm_feature_select=True,
+)
+
+
+# ## Perform extraction with PyBaSiC CellProfiler method
+# 
+# - Merge single cells
+# - Normalize
+# - Feature selection on normalized features
+
+# In[7]:
+
+
+sc_util.extract_single_cells(
+    single_cell_file=single_cell_file2,
+    linking_cols=linking_cols,
+    platemap_df=platemap_df,
+    output_folder=output_dir,
+    method_name=method2,
+    norm_feature_select=True,
+)
+
+
+# ## Perform extraction with PyBaSiC Cellpose method
+# 
+# - Merge single cells
+# - Normalize
+# - Feature selection on normalized features
+
+# In[8]:
+
+
+sc_util.extract_single_cells(
+    single_cell_file=single_cell_file3,
+    linking_cols=linking_cols,
+    platemap_df=platemap_df,
+    output_folder=output_dir,
+    method_name=method3,
+    norm_feature_select=True,
+)
+
+
+# ## Perform extraction with CellProfiler Cellpose method
+# 
+# - Merge single cells
+# - Normalize
+# - Feature selection on normalized features
+
+# In[9]:
+
+
+sc_util.extract_single_cells(
+    single_cell_file=single_cell_file4,
+    linking_cols=linking_cols,
+    platemap_df=platemap_df,
+    output_folder=output_dir,
+    method_name=method4,
+    norm_feature_select=True,
+)
+
+
+# ---
+# ## Visualize count statisitics
+
+# ### All CellProfiler
+
+# In[10]:
+
+
+data_path = f"{output_dir}/nf1_sc_{method1}.csv.gz"
+data_df = pd.read_csv(data_path, compression="gzip")
+
+data_df.Metadata_genotype.value_counts()
+
+
+# In[11]:
+
+
+pd.crosstab(data_df.Metadata_genotype, data_df.Metadata_Well)
```

Comment:
Consider adding a comment to explain what this function does.

Reply:
^Given your use of `crosstab` throughout this file, I think it would be enough to add a comment here and nowhere else in this file.

Reply:
Added in commit [e2512de](https://github.com/WayScience/NF1_SchwannCell_data/pull/42/commits/e2512de1d4f51232a14dc02f74361e09342550a0)